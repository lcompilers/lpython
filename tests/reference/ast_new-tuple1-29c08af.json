{
    "basename": "ast_new-tuple1-29c08af",
    "cmd": "lpython --show-ast --new-parser --no-color {infile} -o {outfile}",
    "infile": "tests/parser/tuple1.py",
    "infile_hash": "a598d163f33a7eb37376592c8018780cb4024618f4a8024bd728ff15",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast_new-tuple1-29c08af.stdout",
    "stdout_hash": "ba3dcb7011c5466eb869001acbc2c1a89c60f15e59dcdf8e7974baab",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}