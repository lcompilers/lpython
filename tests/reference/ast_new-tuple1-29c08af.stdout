(Module
    [(Assign
        [(Name
            a
            Store
        )]
        (BinOp
            (Tuple
                [(ConstantInt
                    1
                    ()
                )]
                Load
            )
            Add
            (ConstantInt
                4
                ()
            )
        )
        ()
    )
    (Assign
        [(Name
            a
            Store
        )]
        (BinOp
            (BinOp
                (ConstantInt
                    3
                    ()
                )
                Mult
                (Tuple
                    [(ConstantInt
                        1
                        ()
                    )
                    (ConstantInt
                        2
                        ()
                    )]
                    Load
                )
            )
            Add
            (ConstantInt
                4
                ()
            )
        )
        ()
    )
    (Expr
        (Call
            (Name
                f
                Load
            )
            [(Tuple
                [(Name
                    x
                    Load
                )
                (Name
                    y
                    Load
                )]
                Load
            )]
            []
        )
    )
    (Expr
        (Call
            (Name
                f
                Load
            )
            [(Tuple
                [(Name
                    x
                    Load
                )]
                Load
            )]
            []
        )
    )
    (Assign
        [(Name
            a
            Store
        )]
        (Call
            (Name
                len
                Load
            )
            [(Tuple
                [(ConstantInt
                    1
                    ()
                )
                (ConstantInt
                    2
                    ()
                )
                (ConstantInt
                    3
                    ()
                )]
                Load
            )]
            []
        )
        ()
    )
    (Assign
        [(Name
            a4
            Store
        )]
        (Tuple
            [(Tuple
                [(ConstantInt
                    1
                    ()
                )
                (ConstantInt
                    2
                    ()
                )
                (ConstantInt
                    3
                    ()
                )]
                Load
            )
            (Tuple
                [(ConstantInt
                    4
                    ()
                )
                (ConstantInt
                    5
                    ()
                )
                (ConstantInt
                    6
                    ()
                )]
                Load
            )]
            Load
        )
        ()
    )
    (Expr
        (Tuple
            [(ConstantInt
                1
                ()
            )
            (ConstantInt
                2
                ()
            )
            (ConstantInt
                3
                ()
            )]
            Load
        )
    )
    (Expr
        (Tuple
            [(ConstantInt
                1
                ()
            )
            (ConstantInt
                2
                ()
            )]
            Load
        )
    )
    (Expr
        (BinOp
            (Tuple
                [(ConstantInt
                    1
                    ()
                )
                (ConstantInt
                    2
                    ()
                )]
                Load
            )
            Mult
            (ConstantInt
                3
                ()
            )
        )
    )
    (Expr
        (Call
            (Name
                print
                Load
            )
            [(BinOp
                (ConstantStr
                    "%s%s%s"
                    ()
                )
                Mod
                (Tuple
                    [(Name
                        a
                        Load
                    )
                    (Name
                        b
                        Load
                    )
                    (Name
                        c
                        Load
                    )]
                    Load
                )
            )]
            []
        )
    )
    (Expr
        (Call
            (Name
                print
                Load
            )
            [(BinOp
                (ConstantStr
                    "%s%s%s"
                    ()
                )
                Mod
                (Tuple
                    [(BinOp
                        (ConstantInt
                            10
                            ()
                        )
                        Add
                        (Name
                            x
                            Load
                        )
                    )
                    (Name
                        y
                        Load
                    )
                    (BinOp
                        (Name
                            z
                            Load
                        )
                        Add
                        (ConstantInt
                            20
                            ()
                        )
                    )]
                    Load
                )
            )]
            []
        )
    )
    (If
        (Compare
            (Attribute
                (Name
                    sys
                    Load
                )
                version_info
                Load
            )
            GtE
            [(Tuple
                [(ConstantInt
                    3
                    ()
                )
                (ConstantInt
                    9
                    ()
                )]
                Load
            )]
        )
        [(Pass)]
        []
    )
    (Assign
        [(Name
            NDArray
            Store
        )]
        (Call
            (Name
                _GenericAlias
                Load
            )
            [(Attribute
                (Name
                    np
                    Load
                )
                ndarray
                Load
            )
            (Tuple
                [(Name
                    Any
                    Load
                )
                (Name
                    _DType
                    Load
                )]
                Load
            )]
            []
        )
        ()
    )
    (FunctionDef
        test
        ([]
        []
        []
        []
        []
        []
        [])
        [(Pass)]
        [(Call
            (Attribute
                (Attribute
                    (Name
                        pytest
                        Load
                    )
                    mark
                    Load
                )
                parametrize
                Load
            )
            [(ConstantStr
                "shape"
                ()
            )
            (List
                [(ConstantInt
                    5
                    ()
                )
                (Tuple
                    [(ConstantInt
                        3
                        ()
                    )
                    (ConstantInt
                        3
                        ()
                    )
                    (ConstantInt
                        3
                        ()
                    )]
                    Load
                )]
                Load
            )]
            []
        )]
        ()
        ()
    )
    (Assign
        [(Name
            _promotion_table
            Store
        )]
        (Dict
            [(Tuple
                [(Name
                    int8
                    Load
                )
                (Name
                    int8
                    Load
                )]
                Load
            )]
            [(Name
                int8
                Load
            )]
        )
        ()
    )
    (Expr
        (Tuple
            [(Name
                x
                Load
            )
            (Name
                y
                Load
            )]
            Load
        )
    )
    (Expr
        (Tuple
            [(Name
                x
                Load
            )
            (Name
                y
                Load
            )]
            Load
        )
    )
    (Expr
        (Tuple
            [(Call
                (Name
                    assert_
                    Load
                )
                [(Call
                    (Attribute
                        (Name
                            np
                            Load
                        )
                        isnan
                        Load
                    )
                    [(Name
                        div
                        Load
                    )]
                    []
                )]
                []
            )
            (BinOp
                (ConstantStr
                    "dt: %s, rem: %s"
                    ()
                )
                Mod
                (Tuple
                    [(Name
                        dt
                        Load
                    )
                    (Name
                        rem
                        Load
                    )]
                    Load
                )
            )]
            Load
        )
    )
    (Expr
        (Call
            (Name
                sum
                Load
            )
            [(ListComp
                (Subscript
                    (Tuple
                        [(ConstantInt
                            1
                            ()
                        )
                        (UnaryOp
                            USub
                            (ConstantInt
                                1
                                ()
                            )
                        )]
                        Load
                    )
                    (BinOp
                        (Name
                            i
                            Load
                        )
                        Mod
                        (ConstantInt
                            2
                            ()
                        )
                    )
                    Load
                )
                [((Name
                    i
                    Store
                )
                (Name
                    x
                    Load
                )
                []
                0)]
            )]
            []
        )
    )
    (Expr
        (Call
            (Name
                sum
                Load
            )
            [(ListComp
                (Subscript
                    (Tuple
                        [(ConstantInt
                            1
                            ()
                        )
                        (UnaryOp
                            USub
                            (ConstantInt
                                1
                                ()
                            )
                        )]
                        Load
                    )
                    (BinOp
                        (Name
                            i
                            Load
                        )
                        Mod
                        (ConstantInt
                            2
                            ()
                        )
                    )
                    Load
                )
                [((Name
                    i
                    Store
                )
                (Name
                    x
                    Load
                )
                []
                0)]
            )]
            []
        )
    )
    (Assign
        [(Name
            a
            Store
        )]
        (Tuple
            [(Name
                x
                Load
            )
            (Name
                y
                Load
            )]
            Load
        )
        ()
    )
    (AnnAssign
        (Name
            t
            Store
        )
        (Subscript
            (Name
                tuple
                Load
            )
            (Tuple
                [(Name
                    i32
                    Load
                )
                (Name
                    i32
                    Load
                )]
                Load
            )
            Load
        )
        (Tuple
            [(ConstantInt
                2
                ()
            )
            (ConstantInt
                3
                ()
            )]
            Load
        )
        1
    )]
    [(TypeIgnore
        0
        ""
    )]
)
